packages/cli/src/commands/worker.ts

```diff
@@ -483,6 +483,19 @@ export class Worker extends BaseCommand {
 			await this.setupHealthMonitor();
 		}
 
+		if (process.stdout.isTTY) {
+			process.stdin.setRawMode(true);
+			process.stdin.resume();
+			process.stdin.setEncoding('utf8');
+
+			process.stdin.on('data', (key: string) => {
+				// ctrl+c
+				if (key.charCodeAt(0) === 3) {
+					void this.stopProcess();
+				}
+			});
+		}
```

Comment:
could this be moved to `BaseCommand.init` ?

Reply:
As far as I understand, I think this is only needed for `worker` and `webhook` because of their [never-resolving promise](https://github.com/n8n-io/n8n/pull/5461/files). `start` has its own `process.stdin` listener and I'd rather not change `webhook` in this PR about workers.